numpy/core/code_generators/generate_umath.py

```diff
@@ -864,6 +864,20 @@ def english_upper(s):
           TypeDescription('d', None, 'd', 'di'),
           TypeDescription('g', None, 'g', 'gi'),
           ],
+          ),
+'gcd' :
+    Ufunc(2, 1, Zero,
+          docstrings.get('numpy.core.umath.gcd'),
+          "PyUFunc_SimpleBinaryOperationTypeResolver",
+          TD(ints),
+          TD('O', f='npy_ObjectGCD'),
+          ),
+'lcm' :
+    Ufunc(2, 1, None,
```

Comment:
Got this wrong until now. The identity of gcd is `0` since  by definition, `gcd(a, 0) == a`.

The identity for `lcm` is some kind of epsilon. For the integers, that epsilon is simply `1`. However, for the rationals, (ie `decimal.Decimal`), this is incorrect, nor is such an epsilon really representable.